File: qe-selenium-java/Activity8.java
package demos;

import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.firefox.FirefoxDriver;
import org.openqa.selenium.interactions.Actions;

public class Activity8 {
public static void main(String[] args) {
    WebDriver driver = new FirefoxDriver();
    Actions builder = new Actions(driver);

    driver.get("https://training-support.net/webelements/mouse-events");
    // WebDriverWait wait = new WebDriverWait(driver, Duration.ofSeconds(10));
    
    System.out.println(driver.getTitle());

    WebElement cargoLockButton = driver.findElement(By.cssSelector("div.file:nth-child(1) > div:nth-child(1)"));
    WebElement  cargoTomlButton= driver.findElement(By.cssSelector("div.file:nth-child(2) > div:nth-child(1)"));
  
    WebElement srcButton = driver.findElement(By.cssSelector("div.file:nth-child(3) > div:nth-child(1)"));
    WebElement targetButton = driver.findElement(By.cssSelector("div.file:nth-child(4) > div:nth-child(1)"));


         builder
               .click(cargoLockButton)
               .moveToElement(cargoTomlButton)
               .click(cargoTomlButton)
               .build()
              .perform();
              

WebElement messageElement = driver.findElement(By.id("result"));
String message1 = messageElement.getText();
System.out.println(message1);

// Double click on src
builder.doubleClick(srcButton).perform();

String message2 = messageElement.getText();
System.out.println(message2);

// Right click on target
builder.contextClick(targetButton).perform();

WebElement openButton = driver.findElement(By.cssSelector("li.w-full:nth-child(1)"));

builder.click(openButton).perform();

try {
    Thread.sleep(500);
} catch (InterruptedException ex) {
    ex.printStackTrace();
}

String openBtnMessage = messageElement.getText();
System.out.println(openBtnMessage);

driver.quit();

}
}

File: qe-selenium-python/Activity8.py
import time
from selenium import webdriver
from selenium.webdriver.common.by import By
from selenium.webdriver.support.ui import WebDriverWait
from selenium.webdriver.support import expected_conditions as EC


with webdriver.Firefox() as driver:
    actions = webdriver.ActionChains(driver)
    wait = WebDriverWait(driver,20)
    driver.get('https://training-support.net/webelements/mouse-events')
    print(driver.title)

    cargo_lock_btn = driver.find_element(By.CSS_SELECTOR, 'div.file:nth-child(1) > div:nth-child(1)')
    cargo_toml_btn = driver.find_element(By.CSS_SELECTOR, 'div.file:nth-child(2) > div:nth-child(1)')
    src_btn = driver.find_element(By.CSS_SELECTOR, 'div.file:nth-child(3) > div:nth-child(1)')
    target_btn = driver.find_element(By.CSS_SELECTOR, 'div.file:nth-child(4) > div:nth-child(1)')

    actions.click(cargo_lock_btn).move_to_element(cargo_toml_btn).click(cargo_toml_btn).perform()

    msg = driver.find_element(By.ID, 'result').text

    print(msg)

    actions.double_click(src_btn).context_click(target_btn).perform()

    open_btn = driver.find_element(By.CSS_SELECTOR,'li.w-full:nth-child(1)')

    actions.click(open_btn).perform()

    wait.until(lambda d: len(d.find_element(By.ID, 'result').text) != len(msg))

    print(driver.find_element(By.ID,'result').text)

    driver.quit()
